src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -42,6 +42,31 @@
 
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  // PROTO Data structure:
+  // Parse the contents of graph.txt into a proto Graph object, and extract
+  // information
+  // from the proto object into a map. This is used to store the proto input and
+  // isn't updated
+  // with mutations.
+  private Graph protoGraph = null;
```

Comment:
Do we need to store protoGraph and protoNodesMap? Seems like these two are only used when we parse the graph. All the other places only consume graphOriginal and graph.

Reply:
yea you're right, we don't. thanks for pointing it out!